clients/unshaded/src/main/java/tachyon/client/TachyonFile.java

```diff
@@ -497,7 +497,7 @@ boolean recache(int blockIndex) throws IOException {
       int bufferBytes =
           (int) mTachyonConf.getBytes(Constants.USER_FILE_BUFFER_BYTES, Constants.MB) * 4;
       byte[] buffer = new byte[bufferBytes];
-      bos = new BlockOutStream(this, WriteType.TRY_CACHE, blockIndex, mTachyonConf);
+      bos = new LocalBlockOutStream(this, WriteType.TRY_CACHE, blockIndex, mTachyonConf);
```

Comment:
should this be `LocalBlockOutStream` always? If `!mTachyonFS.hasLocalWorker()`, `new LocalBlockOutStream` will throw exceptions